main_test.go

```diff
@@ -100,3 +100,47 @@ func TestDiscoveryWithMockSources(t *testing.T) {
 
 	})
 }
+
+func TestAddLabels(t *testing.T) {
```

Comment:
This should change appropriately as we are going to split remove and add into two functions.